conda/core/link.py

```diff
@@ -237,15 +241,37 @@ def verify(self):
                     rm_rf(self.transaction_context['temp_dir'])
                     raise
                 log.info(exceptions)
-
+        try:
+            self._verify_pre_link_message(
+                itertools.chain(
+                    *(act.link_action_groups for act in self.prefix_action_groups.values())
+                )
+            )
+        except CondaSystemExit:
+            rm_rf(self.transaction_context['temp_dir'])
+            raise
         self._verified = True
 
+    def _verify_pre_link_message(self, all_link_groups):
+        flag_pre_link = False
+        for act in all_link_groups:
+            prelink_msg_dir = (
+                    Path(act.pkg_data.extracted_package_dir) / "info" / "prelink_messages"
+            )
+            if prelink_msg_dir.is_dir() and list(prelink_msg_dir.glob("**/*")):
+                print(f"\nPre-link message from {act.pkg_data.repodata_record}:")
```

Comment:
Should this be using a logger instead of printing to stdout?

Reply:
sure, should I change it to ``log.info``?

Reply:
I changed it to ``log.info``

Reply:
Yeah, that makes sense. 👍🏻